homeassistant/components/blink/__init__.py

```diff
@@ -1,153 +1,161 @@
-
 SERVICE_SAVE_VIDEO_SCHEMA = vol.Schema(
     {vol.Required(CONF_NAME): cv.string, vol.Required(CONF_FILENAME): cv.string}
 )
+SERVICE_SEND_PIN_SCHEMA = vol.Schema({vol.Optional(CONF_PIN): cv.string})
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             {
                 vol.Required(CONF_USERNAME): cv.string,
                 vol.Required(CONF_PASSWORD): cv.string,
-                vol.Optional(
-                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
-                ): cv.time_period,
-                vol.Optional(CONF_BINARY_SENSORS, default={}): BINARY_SENSOR_SCHEMA,
-                vol.Optional(CONF_SENSORS, default={}): SENSOR_SCHEMA,
-                vol.Optional(CONF_OFFSET, default=1): int,
-                vol.Optional(CONF_MODE, default=""): cv.string,
+                vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): int,
             }
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-def setup(hass, config):
-    """Set up Blink System."""
+async def async_setup(hass, config):
+    """Set up a config entry."""
+    if DOMAIN not in config:
+        return True
 
-    conf = config[BLINK_DATA]
+    conf = config.get(DOMAIN, {})
     username = conf[CONF_USERNAME]
     password = conf[CONF_PASSWORD]
     scan_interval = conf[CONF_SCAN_INTERVAL]
-    is_legacy = bool(conf[CONF_MODE] == "legacy")
-    motion_interval = conf[CONF_OFFSET]
-    hass.data[BLINK_DATA] = blinkpy.Blink(
+    hass.data[DOMAIN] = blinkpy.Blink(
         username=username,
         password=password,
-        motion_interval=motion_interval,
-        legacy_subdomain=is_legacy,
+        motion_interval=DEFAULT_OFFSET,
+        legacy_subdomain=False,
+        no_prompt=True,
+        device_id="Home Assistant",
     )
-    hass.data[BLINK_DATA].refresh_rate = scan_interval.total_seconds()
-    hass.data[BLINK_DATA].start()
+    hass.data[DOMAIN].refresh_rate = scan_interval
+    hass.data[BLINK_CONFIG] = conf
+    await hass.async_add_executor_job(hass.data[DOMAIN].start)
+
+    if not hass.config_entries.async_entries(DOMAIN) and hass.data[DOMAIN]:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}
+            )
+        )
 
-    platforms = [
-        ("alarm_control_panel", {}),
-        ("binary_sensor", conf[CONF_BINARY_SENSORS]),
-        ("camera", {}),
-        ("sensor", conf[CONF_SENSORS]),
-    ]
+    return True
 
-    for component, schema in platforms:
-        discovery.load_platform(hass, component, DOMAIN, schema, config)
 
-    def trigger_camera(call):
+async def async_setup_entry(hass, entry):
+    """Set up Blink via config entry."""
+    if DOMAIN not in hass.data:
+        username = entry.data[CONF_USERNAME]
+        password = entry.data[CONF_PASSWORD]
+        scan_interval = entry.data[CONF_SCAN_INTERVAL]
+        hass.data[DOMAIN] = blinkpy.Blink(
```

Comment:
key it by config entry so you support multiple accounts.
```suggestion
        hass.data[DOMAIN][entry.entry_id] = blinkpy.Blink(
```